sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -330,6 +331,18 @@ public void testGroupBy() {
       Encoders.STRING());
 
     Assert.assertEquals(asSet("1a", "3foobar"), toSet(flatMapped.collectAsList()));
+    Dataset<String> flatMapSorted = grouped.flatMapSortedGroups(
+        JavaConverters.iterableAsScalaIterable(Collections.singletonList(ds.col("value"))).toSeq(),
+        (FlatMapGroupsFunction<Integer, String, String>) (key, values) -> {
```

Comment:
interesting, so the curried Scala method becomes normal method in Java?

Reply:
Yes, Scala `flatMapSortedGroups(other)(leftOrder)(rightOrder)(func)(implicit encoder)`
in Java becomes `flatMapSortedGroups(other, leftOrder, rightOrder, func, encoder)`.